src/Font/UiFontDescriptor.cc

```diff
@@ -0,0 +1,57 @@
+#include "../../ui.h"
+#include "../ui-node.h"
+#include "nbind/nbind.h"
+
+UiFontDescriptor::UiFontDescriptor(uiFontDescriptor * desc) {
+	d = desc;
+	cleanup = 1;
+}
+
+UiFontDescriptor::UiFontDescriptor(char *family, double size, int weight, int italic, int stretch) {
+    d = new uiFontDescriptor();
+    d->Family = family;
+    d->Size = size;
+    d->Weight = weight;
+    d->Italic = italic;
+    d->Stretch = stretch;
+}
+
+void UiFontDescriptor::free() {
+	if(cleanup){
+		uiFreeFontButtonFont(d);
+	}
+	delete d;
+}
+
+char *UiFontDescriptor::getFamily() {
+	return d->Family;
+}
+
+double UiFontDescriptor::getSize() {
+	return d->Size;
+}
+
+int UiFontDescriptor::getWeight() {
+	return d->Weight;
+}
+
+int UiFontDescriptor::getItalic() {
+	return d->Italic;
+}
+
+int UiFontDescriptor::getStretch() {
+	return d->Stretch;
+}
+
+uiFontDescriptor *UiFontDescriptor::getHandle(){
+	return d;
+}
+
+
+NBIND_CLASS(UiFontDescriptor) {
+	method(getFamily);
```

Comment:
You should declare constructor here if this class has to be created on js land.

Reply:
`construct<char *, double, int, int, int>();` ?

<details>
<summary>Error message:</summary>
<code><pre>

make: Entering directory '/Users/niklas/Desktop/DEV-TEST/libui/libui-node/build'
  CXX(target) Release/obj.target/nbind/src/Font/UiFontDescriptor.o
In file included from ../src/Font/UiFontDescriptor.cc:2:
In file included from ../src/Font/../ui-node.h:6:
In file included from ../node_modules/nbind/include/nbind/api.h:77:
../node_modules/nbind/include/nbind/v8/ArgFromWire.h:26:32: error: no member named 'fromWireType' in 'nbind::BindingType<char *>'
                return(Transformed::Binding::fromWireType(args[Index]));
                                             ^
../node_modules/nbind/include/nbind/v8/Creator.h:17:50: note: in instantiation of function template specialization 'nbind::ArgFromWire<nbind::PolicyListType<>, 0, char
      *>::get<Nan::FunctionCallbackInfo<v8::Value> >' requested here
                BindWrapper<Bound>::createObj(args, Args(args).get(args)...);
                                                               ^
../node_modules/nbind/include/nbind/signature/ConstructorSignature.h:42:21: note: in instantiation of member function 'nbind::Creator<UiFontDescriptor,
      nbind::TypeList<nbind::ArgFromWire<nbind::PolicyListType<>, 0, char *>, nbind::ArgFromWire<nbind::PolicyListType<>, 1, double>,
      nbind::ArgFromWire<nbind::PolicyListType<>, 2, int>, nbind::ArgFromWire<nbind::PolicyListType<>, 3, int>, nbind::ArgFromWire<nbind::PolicyListType<>, 4, int> >
      >::create' requested here
                ConstructWrapper::create(args);
                                  ^
../node_modules/nbind/include/nbind/signature/BaseSignature.h:84:40: note: in instantiation of member function 'nbind::ConstructorSignature<UiFontDescriptor,
      nbind::PolicyListType<>, char *, double, int, int, int>::call' requested here
                reinterpret_cast<funcPtr>(Signature::call),
                                                     ^
../node_modules/nbind/include/nbind/signature/ConstructorSignature.h:18:2: note: in instantiation of member function
      'nbind::TemplatedBaseSignature<nbind::ConstructorSignature<UiFontDescriptor, nbind::PolicyListType<>, char *, double, int, int, int>, nbind::PolicyListType<>,
      UiFontDescriptor *, char *, double, int, int, int>::TemplatedBaseSignature' requested here
        ConstructorSignature() {
        ^
../node_modules/nbind/include/nbind/signature/BaseSignature.h:95:20: note: in instantiation of member function 'nbind::ConstructorSignature<UiFontDescriptor,
      nbind::PolicyListType<>, char *, double, int, int, int>::ConstructorSignature' requested here
                static Signature instance;
                                 ^
../node_modules/nbind/include/nbind/BindDefiner.h:172:40: note: in instantiation of member function
      'nbind::TemplatedBaseSignature<nbind::ConstructorSignature<UiFontDescriptor, nbind::PolicyListType<>, char *, double, int, int, int>, nbind::PolicyListType<>,
      UiFontDescriptor *, char *, double, int, int, int>::getInstance' requested here
                bindClass.addConstructor(&Signature::getInstance());
                                                     ^
../src/Font/UiFontDescriptor.cc:52:2: note: in instantiation of function template specialization 'nbind::BindDefiner<UiFontDescriptor>::constructor<char *, double, int,
      int, int>' requested here
        construct<char *, double, int, int, int>();
        ^
../node_modules/nbind/include/nbind/nbind.h:16:19: note: expanded from macro 'construct'
#define construct NBIND_CONSTRUCT
                  ^
../node_modules/nbind/include/nbind/noconflict.h:70:33: note: expanded from macro 'NBIND_CONSTRUCT'
#define NBIND_CONSTRUCT definer.constructor
                                ^
../src/Font/UiFontDescriptor.cc:51:1: note: in instantiation of member function 'BindInvokerUiFontDescriptor<UiFontDescriptor>::BindInvokerUiFontDescriptor' requested here
NBIND_CLASS(UiFontDescriptor) {
^
../node_modules/nbind/include/nbind/noconflict.h:52:26: note: expanded from macro 'NBIND_CLASS'
#define NBIND_CLASS(...) NBIND_SELECT(NBIND_CLASS, __VA_ARGS__)
                         ^
../node_modules/nbind/include/nbind/noconflict.h:24:33: note: expanded from macro 'NBIND_SELECT'
#define NBIND_SELECT(name, ...) NBIND_SELECT_HELPER(name, NBIND_EXPAND(NBIND_SIZE(__VA_ARGS__)))(__VA_ARGS__)
                                ^
../node_modules/nbind/include/nbind/noconflict.h:23:41: note: expanded from macro 'NBIND_SELECT_HELPER'
#define NBIND_SELECT_HELPER(name, argc) NBIND_CONCAT(name ## _, argc)
                                        ^
note: (skipping 1 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
<scratch space>:344:1: note: expanded from here
NBIND_CLASS_1
^
../node_modules/nbind/include/nbind/noconflict.h:41:40: note: expanded from macro 'NBIND_CLASS_1'
        static struct BindInvoker##Name<Name> bindInvoker##Name; \
                                              ^
<scratch space>:4:1: note: expanded from here
bindInvokerUiFontDescriptor
^
In file included from ../src/Font/UiFontDescriptor.cc:3:
In file included from ../node_modules/nbind/include/nbind/nbind.h:8:
In file included from ../node_modules/nbind/include/nbind/noconflict.h:8:
In file included from ../node_modules/nbind/include/nbind/BindDefiner.h:25:
../node_modules/nbind/include/nbind/v8/Creator.h:21:70: error: no matching member function for call to 'get'
                static_cast<TemplatedArgStorage<Bound> &>(storage).init(Args(args).get(args)...);
                                                                        ~~~~~~~~~~~^~~
../node_modules/nbind/include/nbind/signature/ConstructorSignature.h:46:21: note: in instantiation of member function 'nbind::Creator<UiFontDescriptor,
      nbind::TypeList<nbind::ArgFromWire<nbind::PolicyListType<>, 0, char *>, nbind::ArgFromWire<nbind::PolicyListType<>, 1, double>,
      nbind::ArgFromWire<nbind::PolicyListType<>, 2, int>, nbind::ArgFromWire<nbind::PolicyListType<>, 3, int>, nbind::ArgFromWire<nbind::PolicyListType<>, 4, int> >
      >::createValue' requested here
                ConstructWrapper::createValue(storage, args);
                                  ^
../node_modules/nbind/include/nbind/signature/ConstructorSignature.h:19:55: note: in instantiation of member function 'nbind::ConstructorSignature<UiFontDescriptor,
      nbind::PolicyListType<>, char *, double, int, int, int>::createValue' requested here
                this->setValueConstructor(reinterpret_cast<funcPtr>(createValue));
                                                                    ^
../node_modules/nbind/include/nbind/signature/BaseSignature.h:95:20: note: in instantiation of member function 'nbind::ConstructorSignature<UiFontDescriptor,
      nbind::PolicyListType<>, char *, double, int, int, int>::ConstructorSignature' requested here
                static Signature instance;
                                 ^
../node_modules/nbind/include/nbind/BindDefiner.h:172:40: note: in instantiation of member function
      'nbind::TemplatedBaseSignature<nbind::ConstructorSignature<UiFontDescriptor, nbind::PolicyListType<>, char *, double, int, int, int>, nbind::PolicyListType<>,
      UiFontDescriptor *, char *, double, int, int, int>::getInstance' requested here
                bindClass.addConstructor(&Signature::getInstance());
                                                     ^
../src/Font/UiFontDescriptor.cc:52:2: note: in instantiation of function template specialization 'nbind::BindDefiner<UiFontDescriptor>::constructor<char *, double, int,
      int, int>' requested here
        construct<char *, double, int, int, int>();
        ^
../node_modules/nbind/include/nbind/nbind.h:16:19: note: expanded from macro 'construct'
#define construct NBIND_CONSTRUCT
                  ^
../node_modules/nbind/include/nbind/noconflict.h:70:33: note: expanded from macro 'NBIND_CONSTRUCT'
#define NBIND_CONSTRUCT definer.constructor
                                ^
../src/Font/UiFontDescriptor.cc:51:1: note: in instantiation of member function 'BindInvokerUiFontDescriptor<UiFontDescriptor>::BindInvokerUiFontDescriptor' requested here
NBIND_CLASS(UiFontDescriptor) {
^
../node_modules/nbind/include/nbind/noconflict.h:52:26: note: expanded from macro 'NBIND_CLASS'
#define NBIND_CLASS(...) NBIND_SELECT(NBIND_CLASS, __VA_ARGS__)
                         ^
../node_modules/nbind/include/nbind/noconflict.h:24:33: note: expanded from macro 'NBIND_SELECT'
#define NBIND_SELECT(name, ...) NBIND_SELECT_HELPER(name, NBIND_EXPAND(NBIND_SIZE(__VA_ARGS__)))(__VA_ARGS__)
                                ^
../node_modules/nbind/include/nbind/noconflict.h:23:41: note: expanded from macro 'NBIND_SELECT_HELPER'
#define NBIND_SELECT_HELPER(name, argc) NBIND_CONCAT(name ## _, argc)
                                        ^
note: (skipping 1 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
<scratch space>:344:1: note: expanded from here
NBIND_CLASS_1
^
../node_modules/nbind/include/nbind/noconflict.h:41:40: note: expanded from macro 'NBIND_CLASS_1'
        static struct BindInvoker##Name<Name> bindInvoker##Name; \
                                              ^
<scratch space>:4:1: note: expanded from here
bindInvokerUiFontDescriptor
^
../node_modules/nbind/include/nbind/v8/ArgFromWire.h:25:36: note: candidate template ignored: substitution failure [with NanArgs = Nan::FunctionCallbackInfo<v8::Value>]
        inline typename Transformed::Type get(const NanArgs &args) noexcept(false) {
                                          ^


</pre></code>
</details>

Reply:
It should work AFAIK. @jjrv could you kindly help us here?

Reply:
UiLabel has similar constructor, but the argument is `const char *`. Could you try to change it to const?
https://github.com/parro-it/libui-node/blob/957a61d99775a3922ab8fcc5b2c72670012caf2f/src/UiLabel.cc

Reply:
Yes, const solves the nbind error.